src/modlunit/units.c

```diff
@@ -1014,13 +1014,63 @@ static void init()
 	goto l0;
 }
 
+#if NRN_DYNAMIC_UNITS
+static double modern_getflt() {
+	int c;
```

Comment:
@nrnhines : you mentioned about possibly reducing code duplication?

Reply:
I was thinking in terms of just the calculation of d and kept the lexical analysis. Of course it could be reduced further but I didn't think it was worth the trouble.

Reply:
ok